x-pack/legacy/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/classification_exploration/evaluate_panel.tsx

```diff
@@ -0,0 +1,321 @@
+import { DATA_FRAME_TASK_STATE } from '../../../analytics_management/components/analytics_list/common';
+import {
+  isResultsSearchBoolQuery,
+  isClassificationEvaluateResponse,
+  ConfusionMatrix,
+  ResultsSearchQuery,
+  ANALYSIS_CONFIG_TYPE,
+} from '../../../../common/analytics';
+import { LoadingPanel } from '../loading_panel';
+import { getColumnData } from './column_data';
+
+const defaultPanelWidth = 500;
+
+interface Props {
+  jobConfig: DataFrameAnalyticsConfig;
+  jobStatus: DATA_FRAME_TASK_STATE;
+  searchQuery: ResultsSearchQuery;
+}
+
+export const EvaluatePanel: FC<Props> = ({ jobConfig, jobStatus, searchQuery }) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [confusionMatrixData, setConfusionMatrixData] = useState<ConfusionMatrix[]>([]);
+  const [columns, setColumns] = useState<any>([]);
+  const [columnsData, setColumnsData] = useState<any>([]);
+  const [popoverContents, setPopoverContents] = useState<any>([]);
+  const [docsCount, setDocsCount] = useState<null | number>(null);
+  const [error, setError] = useState<null | string>(null);
+  const [panelWidth, setPanelWidth] = useState<number>(defaultPanelWidth);
+
+  // Column visibility
+  const [visibleColumns, setVisibleColumns] = useState(() =>
+    columns.map(({ id }: { id: string }) => id)
+  );
+
+  const index = jobConfig.dest.index;
+  const dependentVariable = getDependentVar(jobConfig.analysis);
+  const predictionFieldName = getPredictionFieldName(jobConfig.analysis);
+  // default is 'ml'
+  const resultsField = jobConfig.dest.results_field;
+
+  const loadData = async ({
+    isTrainingClause,
+    ignoreDefaultQuery = true,
+  }: {
+    isTrainingClause: { query: string; operator: string };
+    ignoreDefaultQuery?: boolean;
+  }) => {
+    setIsLoading(true);
+
+    const evalData = await loadEvalData({
+      isTraining: false,
+      index,
+      dependentVariable,
+      resultsField,
+      predictionFieldName,
+      searchQuery,
+      ignoreDefaultQuery,
+      jobType: ANALYSIS_CONFIG_TYPE.CLASSIFICATION,
+    });
+
+    const docsCountResp = await loadDocsCount({
+      isTraining: false,
+      searchQuery,
+      resultsField,
+      destIndex: jobConfig.dest.index,
+    });
+
+    if (
+      evalData.success === true &&
+      evalData.eval &&
+      isClassificationEvaluateResponse(evalData.eval)
+    ) {
+      const confusionMatrix =
+        evalData.eval?.classification?.multiclass_confusion_matrix?.confusion_matrix;
+      setError(null);
+      setConfusionMatrixData(confusionMatrix || []);
+      setIsLoading(false);
+    } else {
+      setIsLoading(false);
+      setConfusionMatrixData([]);
+      setError(evalData.error);
+    }
+
+    if (docsCountResp.success === true) {
+      setDocsCount(docsCountResp.docsCount);
+    } else {
+      setDocsCount(null);
+    }
+  };
+
+  const resizeHandler = () => {
```

Comment:
This doesn't seem to be working properly for me. It works when I expand the window, but if I make the window smaller, the width isn't adjusting:

![image](https://user-images.githubusercontent.com/7405507/70640936-ba6ffd00-1c34-11ea-9020-9c023b56ab18.png)

Reply:
Added investigation into this issue to the meta issue https://github.com/elastic/kibana/issues/51310

It's related to the existing difficulties in the DataGrid component but I can timebox some time to work around it

Reply:
I've created a temporary workaround to ensure the datagrid doesn't overflow the flex container and to make the panels more responsive. c78a2936e8b6d04c4c77bf10e500ca87324afeb9